### ISSUE TITLE
vector-output log density functions

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/stan#1697]
Idea from Aki Vehtari on stan-dev:

In Stan 2.8, log density functions like `normal_log()` output the sum of the densities of their components (with appropriate broadcasting of scalars).

It would be nice to allow them to be specified to return a vector, for instance for WAIC or LOO calculations. 

This issue is closely related to #1683.

**Comment by danielinteractive** [comment]
Just found this, this would be really useful in many situations.

**Comment by bob-carpenter** [comment]
+1. There have been more and more cases where I've wanted to do this.  It shouldn't be too hard by decorating the log density functions with one more layer of suffix.

**Comment by SteveBronder** [comment]
Yes I think it's very doable as laid out above, mostly need to find time do it. There's a few places we need to be careful but for the most part a couple template tricks should handle everything

**Comment by WardBrian** [comment]
For me I think my question is whether the <> syntax is the right choice for this? The alternative that comes to my mind is if we add a keyword argument syntax
